src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -510,12 +591,12 @@ private OAuthProxy(
             this.grant = Objects.requireNonNull(grant);
             this.isCorsPreflight = isCorsPreflight;
 
+            assert challengeResponseTimeout >= 0;
             if (expiresAtMillis != EXPIRES_NEVER)
             {
-                final long delay = expiresAtMillis - System.currentTimeMillis();
-
-                this.expiryFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId, TOKEN_EXPIRED_SIGNAL);
-                this.grant.acquire();
```

Comment:
Please restore this line, and remove the corresponding lines from `newInitialStream`.